src/ORM/DataQuery.php

```diff
@@ -447,6 +448,34 @@ public function count()
         return $this->getFinalisedQuery()->count("DISTINCT {$quotedColumn}");
     }
 
+    /**
+     * Return whether this dataquery will have records. This will use `EXISTS` statements in SQL which are more
+     * performant - especially when used in combination with indexed columns (that you're filtering on)
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        // Grab a statement selecting "everything" - the engine shouldn't care what's being selected in an "EXISTS"
+        // statement anyway
+        $statement = $this->getFinalisedQuery(null, true);
+        $statement->setSelect('*');
+
+        // Clear distinct, grouping, order, and having as it's not relevant for an exists query
+        $statement->setDistinct(false);
+        $statement->setOrderBy(null);
+        $statement->setGroupBy(null);
+        $statement->setHaving(null);
+
+        // Wrap the whole thing in an "EXISTS"
+        $sql = 'SELECT EXISTS(' . $statement->sql($params) . ')';
+        $result = DB::prepared_query($sql, $params);
+        $row = $result->first();
+        $result = reset($row);
+
+        return $result === 1 || $result === 't';
```

Comment:
Commit history tells me that this is because PGSQL - if we can’t use `(bool)$result` or similar, could you add a comment explaining what `'t'` represents here?

Reply:
The PG adapter returns `'t'`'s and `'f'`'s for true/false. Although the tests are still failing so I might have made a mistake...

Reply:
Note that I’ve fixed this but it probably requires another minor release of the pgsql module.

Reply:
Yeah I think I'll just return to using `$result === 1` and ignore PostgreSQL. I can update the travis builds to use the new release of the postgres module.

Reply:
I've added a comment. While this is redundant in PostgreSQL 2.2 it's probably no harm to have a little BC for the Postgres users.